django/forms/forms.py

```diff
@@ -287,8 +287,8 @@ def add_error(self, field, error):
         if hasattr(error, "error_dict"):
             if field is not None:
                 raise TypeError(
-                    "The argument `field` must be `None` when the `error` "
-                    "argument contains errors for multiple fields."
+                    "The argument `field` must be `None` when the `error` argument "
+                    "is a ValidationError with an error_dict."
```

Comment:
This is a little too specific, since the value provided for the error argument can also be a plain dict and trigger this error condition (because it only becomes a ValidationError under the hood).